### DISCUSSION TITLE
Warning about: Package containing module 'falconpy' is not listed in the project requirements

### DISCUSSION THREAD

**Discussion by morcef** [CrowdStrike/falconpy#551] [Q&A] [answered]
Hi,

I am building a Poetry project with crowdstrike-falconpy as a dependency. When I try to import from falconpy I get the following error:
```
Package containing module 'falconpy' is not listed in the project requirements
```

Snippet from my pyproject.toml:
```
[tool.poetry.dependencies]
crowdstrike-falconpy = "^1.0.3"
```

I am currently using latest version of PyCharm.
Is there a workaround/solution for this, without using `# noinspection` or `ignore this requirement`?
Have you considered renaming the project or package naming?

**Comment by jshcodes** [comment]
> Hi,
> 
> I am building a Poetry project with crowdstrike-falconpy as a dependency. When I try to import from falconpy I get the following error:
> 
> ```
> Package containing module 'falconpy' is not listed in the project requirements
> ```
That's interesting.  (And another strange PyCharm warning.)

> 
> I am currently using latest version of PyCharm. Is there a workaround/solution for this, without using `# noinspection` or `ignore this requirement`? Have you considered renaming the project or package naming?

The package name happened when a community member decided to deploy a version of FalconPy on their own early last year. (Deployed a repo, then imported this repo's `src` folder as a submodule.) They beat us to the PyPI deployment by a day or so and when asked, did not give us access to the "falconpy" package account. Hence, this package name is unusable to us. 😞 

"noinspection" / "ignore" does not feel like the right way to go unless there are no other options. I'll try to recreate this to hunt down another solution.

**Reply by morcef** [reply]
Thank you, once again :) Looking forward to hear from you about your findings!

**Comment by jshcodes** [comment, marked as answer]
Hi @morcef!

I was able to recreate this issue, and as of yet have not found a viable solution outside of telling PyCharm to ignore the warning.

I played around with generating the requirements.txt (both using poetry and the IDE) and then running installation to my virtual environment to no avail.  Even though the requirement is definitely installed, PyCharm still throws the warning.

I'll keep poking around as I review our other PyCharm related warnings, but this appears to be possibly related to a [bug within PyCharm itself](https://youtrack.jetbrains.com/issue/PY-19485).

**Reply by morcef** [reply]
Hi @jshcodes,

Sorry for the delayed response.

Thank you for linking me to the PyCharm  bug. I dug deeper, and found another issue where they are one-by-one adding links between package names and module names. I requested the addition of a link between crowdstrike-falconpy and falconpy, so hopefully that will remove the annoying warning.
https://youtrack.jetbrains.com/issue/PY-27985#focus=Comments-27-5846610.0-0
